fury/actor.py

```diff
@@ -2498,3 +2498,115 @@ def texture_on_sphere(rgb, theta=60, phi=60, interpolate=True):
     earthActor.SetTexture(atext)
 
     return earthActor
+
+
+def sdf(centers, directions=(1, 0, 0), colors=(255, 0, 0), primitives='torus',
+        scale=1):
+    """Create a SDF primitive based actor
+
+    Parameters
+    ----------
+    centers : ndarray, shape (N, 3)
+        SDF primitive positions
+    colors : ndarray (N,3) or (N, 4) or tuple (3,) or tuple (4,)
+        RGB or RGBA (for opacity) R, G, B and A should be at the range [0, 1]
+    directions : ndarray, shape (N, 3)
+        The orientation vector of the SDF primitive.
+    primitive : String
+        The primitive of choice to be rendered.
+        Options are sphere and torus. Default is torus
+    scale : float
+        The size of the SDF primitive
+
+
+    Returns
+    -------
+    vtkActor
+    """
+
+    prims = {'sphere': 1, 'torus': 2, 'ellipsoid': 3}
+
+    verts, faces = fp.prim_box()
+    repeated = fp.repeat_primitive(verts, faces, centers=centers,
+                                   colors=colors, directions=directions,
+                                   scale=scale)
+
+    rep_verts, rep_faces, rep_colors, rep_centers = repeated
+    box_actor = get_actor_from_primitive(rep_verts, rep_faces, rep_colors)
+
+    vtk_center = numpy_support.numpy_to_vtk(rep_centers, deep=True)
+    vtk_center.SetNumberOfComponents(3)
+    vtk_center.SetName("center")
+    box_actor.GetMapper().GetInput().GetPointData().AddArray(vtk_center)
+
+    if(isinstance(primitives,  (list, tuple, np.ndarray))):
+        primlist = []
+        for prim in primitives:
+    	    primlist.append(prims[prim])
+        rep_prims = np.repeat(primlist, verts.shape[0])
+    else:
+        rep_prims = np.repeat(prims[primitives], rep_centers.shape[0], axis=0)
+
+    vtk_primitive = numpy_support.numpy_to_vtk(rep_prims, deep=True)
+    vtk_primitive.SetNumberOfComponents(1)
+    vtk_primitive.SetName("primitive")
+    box_actor.GetMapper().GetInput().GetPointData().AddArray(vtk_primitive)
+
+    if(isinstance(scale, (list, tuple, np.ndarray))):
+        rep_scales = np.repeat(scale, verts.shape[0])
+    else:
+        rep_scales = np.repeat(scale, rep_centers.shape[0], axis=0)
+
+    vtk_scale = numpy_support.numpy_to_vtk(rep_scales, deep=True)
+    vtk_scale.SetNumberOfComponents(1)
+    vtk_scale.SetName("scale")
+    box_actor.GetMapper().GetInput().GetPointData().AddArray(vtk_scale)
+
+    if isinstance(directions, (list, tuple, np.ndarray)) and \
+            len(directions) == 3:
+        rep_directions = np.repeat(directions, rep_centers.shape[0], axis=0)
+    else:
+        rep_directions = np.repeat(directions, verts.shape[0], axis=0)
```

Comment:
this block is repeated 3 times(directions, scale, primitives). maybe it deserves a local function or a function in `utils.py` (check if this function do not exist first

Reply:
I think it's better to keep it as it is currently because each block performs slightly differently.

Reply:
Ok, fine for the moment, we will see later